### PR TITLE
Support label rotate to pie chart

### LINKED ISSUE #10045: pie图表的label的rotate属性设置不生效，当设置rotate时，显示结果总是rotate='radial'的效果。
### Version
4.1.0-release

### Steps to reproduce
新建pie图表，设置label的totate属性，则无论rotate设置为何值，效果总是radial。

### What is expected?
rotate依据设置的结果进行显示

### What is actually happening?
无论rotate设置为何值，效果总是radial。

### PR DESCRIPTION
**Pull request by susiwen8**
Close #10045

### CODE REVIEW

File: src/chart/pie/labelLayout.js [hunk 1]
```diff
@@ -222,7 +222,7 @@ export default function (seriesModel, r, viewWidth, viewHeight, sum) {
         var font = labelModel.getFont();
 
         var labelRotate = labelModel.get('rotate')
-            ? (dx < 0 ? -midAngle + Math.PI : -midAngle) : 0;
+            ? +labelModel.get('rotate') * (Math.PI / 180) : 0;
```
Comment (Ovilia) [inline src/chart/pie/labelLayout.js]: If `rotate` is not set, `midAngle` should be used.

Comment (Ovilia) [reply src/chart/pie/labelLayout.js]: Note that you should also update ECharts doc at [here (cn)](https://github.com/ecomfe/echarts-doc/blob/gh-pages/cn/option/series/pie.md) and [here (en)](https://github.com/ecomfe/echarts-doc/blob/gh-pages/en/option/series/pie.md).

Comment (susiwen8) [reply src/chart/pie/labelLayout.js]: midAngle is added, and I will update doc

File: src/chart/pie/labelLayout.js [hunk 2]
```diff
@@ -222,7 +222,8 @@ export default function (seriesModel, r, viewWidth, viewHeight, sum) {
         var font = labelModel.getFont();
 
         var labelRotate = labelModel.get('rotate')
-            ? (dx < 0 ? -midAngle + Math.PI : -midAngle) : 0;
+            ? +labelModel.get('rotate') * (Math.PI / 180)
+            : (dx < 0 ? -midAngle + Math.PI : -midAngle);
```
Comment (Ovilia) [inline src/chart/pie/labelLayout.js]: Note that when `rotate` is set to be `true`, it should use the old algorithm to automatically rotate.
So I think a better way is to check if it's number, if so, use the new algorithm, if not, use the old one.

Comment (susiwen8) [reply src/chart/pie/labelLayout.js]: Yes, you are right, I did't notice rotate can be boolean

File: src/chart/pie/labelLayout.js [hunk 3]
```diff
@@ -221,8 +221,15 @@ export default function (seriesModel, r, viewWidth, viewHeight, sum) {
...
-        var labelRotate = labelModel.get('rotate')
-            ? (dx < 0 ? -midAngle + Math.PI : -midAngle) : 0;
+        var labelRotate;
+        var rotate = labelModel.get('rotate');
+        if (typeof rotate === 'number') {
+            labelRotate = rotate * (Math.PI / 180);
+        } else {
+            labelRotate = rotate
+                        ? (dx < 0 ? -midAngle + Math.PI : -midAngle)
+                        : 0;
```
Comment (Ovilia) [inline src/chart/pie/labelLayout.js]: 4 spaces for indent.

File: src/chart/pie/labelLayout.js [hunk 4]
```diff
@@ -221,8 +221,15 @@ export default function (seriesModel, r, viewWidth, viewHeight, sum) {
         }
         var font = labelModel.getFont();
 
-        var labelRotate = labelModel.get('rotate')
-            ? (dx < 0 ? -midAngle + Math.PI : -midAngle) : 0;
+        var labelRotate;
+        var rotate = labelModel.get('rotate');
+        if (typeof rotate === 'number') {
+            labelRotate = rotate * (Math.PI / 180);
+        } else {
```
Comment (Ovilia) [inline src/chart/pie/labelLayout.js]: It should be 
```js
}
else {
```

**Pull request merged**